pkg/forklift-api/webhooks/validating-webhook/admitters/secret-admitter.go

```diff
@@ -72,6 +72,10 @@ func (admitter *SecretAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissio
 }
 
 func (admitter *SecretAdmitter) validateProviderSecret() *admissionv1.AdmissionResponse {
+	if _, ok := admitter.secret.Data["cacert"]; ok && container.InsecureProvider(&admitter.secret) {
+		return webhookutils.ToAdmissionResponseError(fmt.Errorf("received a request to add insecure provider with a CA certificate"))
```

Comment:
can't we just ignore the cert in this case?

Reply:
we can but this assumes the user wanted to get an insecure provider which is not necessarily true. if the input is invalid, we better return an error than trying to interpret what the user wanted to achieve